### PR TITLE
feat: move common files to keymanapp/shared-sites

### PR DESCRIPTION
**Pull request by mcdurdin**
This adds linkage to the new https://github.com/keymanapp/shared-sites repository, including downloading a bootstrap.inc.sh script from there to source all common files.

The first time build.sh is run, it detects that it needs to download and run the bootstrap. You'll see something like this in the log:

```
$ ./build.sh
[bootstrap] Bootstrap required
[bootstrap] Bootstrap starting
[bootstrap]   Downloading bootstrap.inc.sh
[bootstrap] Downloading _common files
[bootstrap]   Downloading _common/builder.inc.sh
[bootstrap]   Downloading _common/docker.inc.sh
[bootstrap]   Downloading _common/keyman-local-ports.inc.sh
[bootstrap]   Downloading _common/JsonApiFailure.php
[bootstrap]   Downloading _common/KeymanHosts.php
[bootstrap]   Downloading _common/KeymanSentry.php
[bootstrap]   Downloading _common/MarkdownHost.php
[bootstrap] All _common files downloaded
[bootstrap] Bootstrap complete
```

If you want to re-download, you can change the `BOOTSTRAP_VERSION` variable or call `./build.sh configure`. 

Caching note: raw.githubusercontent.com caches files on first retrieval. This means re-downloading files from github may get a stale version of a file if the `BOOTSTRAP_VERSION` variable is set to a branch or tag which has recently changed. Conversely, if it is set to a commit, then you are guaranteed to get the version in the commit, of course.

This goes together with https://github.com/keymanapp/shared-sites/pull/1. 

## TODO Before merge

Note that once https://github.com/keymanapp/shared-sites/pull/1 merges, we'll need to create a tag on shared-sites and refresh the version in this PR accordingly.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -6,20 +6,18 @@ set -eu
...
 readonly HELP_CONTAINER_NAME=help-keyman-website
 readonly HELP_CONTAINER_DESC=help-keyman-com-app
 readonly HELP_IMAGE_NAME=help-keyman-website
 readonly HOST_HELP_KEYMAN_COM=help.keyman.com.localhost
 
-. "$REPO_ROOT/_common/keyman-local-ports.inc.sh"
-. "$REPO_ROOT/_common/docker.inc.sh"
-
-# This script runs from its own folder
-cd "$REPO_ROOT"
```
Comment (mcdurdin) [inline build.sh:22]: This has been moved to bootstrap.inc.sh

File: build.sh [hunk 2]
```diff
@@ -6,20 +6,18 @@ set -eu
 
 ## START STANDARD BUILD SCRIPT INCLUDE
 # adjust relative paths as necessary
-THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
-. "$(dirname "$THIS_SCRIPT")/resources/builder.inc.sh"
+readonly THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+readonly BOOTSTRAP="$(dirname "$THIS_SCRIPT")/resources/bootstrap.inc.sh"
+[ -f "$BOOTSTRAP" ] && source "$BOOTSTRAP" || source <(curl -fs https://raw.githubusercontent.com/keymanapp/shared-sites/main/bootstrap.inc.sh?token=$(date +%s))
 ## END STANDARD BUILD SCRIPT INCLUDE
```
Comment (mcdurdin) [inline build.sh:12]: Note the new prolog which is specific to the keyman sites.

File: build.sh [hunk 3]
```diff
@@ -1,25 +1,19 @@
 #!/usr/bin/env bash
-#
-# Setup help.keyman.com site to run via Docker.
-#
-set -eu
-
```
Comment (mcdurdin) [inline build.sh:6]: `set -eu` moved to bootstrap.inc.sh. We'll rely on the `builder_describe` line below to give detail about what the file does, to DRY this script further.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Pull request merged**